### PR TITLE
Add documentation for 'inheriting' from the test decorator

### LINKED ISSUE #4160: Document how to "inherit from" cocotb.test
Document how to extend the @cocotb.test decorator, based on https://github.com/pyuvm/pyuvm/blob/master/pyuvm/extension_classes.py

### PR DESCRIPTION
**Pull request by ktbarrett**
Closes #4160.

### CODE REVIEW

File: src/cocotb/_decorators.py [hunk 1]
```diff
@@ -373,6 +373,25 @@ async def my_test(dut):
...
         The test function to which the decorator is applied.
+
+    .. note::
+
+        To extend the test decorator, use the following template to create a new
+        ``cocotb.test``\-like wrapper.
+
+        .. code-block:: python3
+
+            def test_extender(**decorator_kwargs):
```
Comment (cmarqu) [inline src/cocotb/_decorators.py:384]: Maybe add "import functools" for the code to be self-contained.

**Review by cmarqu — APPROVED** [review]

File: src/cocotb/_decorators.py (hunk 1 above)
Comment (ktbarrett) [inline src/cocotb/_decorators.py:384]:
```suggestion
            import functools

            def test_extender(**decorator_kwargs):
```

File: src/cocotb/_decorators.py [hunk 2]
```diff
@@ -373,6 +373,27 @@ async def my_test(dut):
...
+
+    .. note::
+
+        To extend the test decorator, use the following template to create a new
+        ``cocotb.test``\-like wrapper.
+
+        .. code-block:: python3
+
+            import functools
+
```
Comment (ktbarrett) [inline src/cocotb/_decorators.py:385]:
```suggestion


```

**Review by imphil — APPROVED** [review]
Thanks!

**Pull request merged**